backend/api/views/TimeRecordView.py

```diff
@@ -0,0 +1,47 @@
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+from api.models import TimeRecord, EventType
+from api.serializers.TimeRecordSerializer import TimeRecordSerializer
+from drf_spectacular.utils import extend_schema, OpenApiParameter
+from django.db.models import Q
+from django.db.models.functions import Cast
+from django.db.models.functions import Collate
+from rest_framework.exceptions import ValidationError
+from rest_framework import status
+
+import logging
+
+logger = logging.getLogger('django')
+
+
+@extend_schema(tags=['Time Record'], request=TimeRecordSerializer)
+class TimeRecordViewSet(viewsets.ModelViewSet):
+    queryset = TimeRecord.objects.all()
+    serializer_class = TimeRecordSerializer
+    permission_classes = [IsAuthenticated]
+    http_method_names = ['get', 'post', 'delete', 'patch']
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        athlete_name = self.request.query_params.get('athlete_name', None)
+        event_type_id = self.request.query_params.get('event_type_id', None)
```

Comment:
We might want to consider filtering by group, as swim meets often involve a combination of group and event_type.